thunder/dynamo/splitter.py

```diff
@@ -140,20 +144,40 @@ def is_thunder_supported_partition(node: torch.fx.Node) -> bool:
     # Call compile on the split region/s.
     thunder_compiled_fns = []
     submodule_to_compiled_fns = {}
+
+    # If the split_gm contains Thunder-traceable checkpoint operators, the checkpointed submodules are replaced, so we need to record the original split_gm
+    origin_split_gm = split_gm
+    if has_thunder_tracable_checkpoint:
+        import copy
+
+        split_gm = copy.deepcopy(split_gm)
```

Comment:
How it works with splitter:
when checkpoint operator is contained in `submod_1`, the `split_gm` is represented as follows:
```
GraphModule(
  (wrap_body_0): GraphModule() # the module checkpoint calls
  (submod_1): GraphModule() # the module contains the checkpoint operator
)
```
The converter will keep the original split gm as `original_split_gm` and deepcopy the `split_gm`, then replace the torch operaters in the submodule `wrap_body_0` of `split_gm`, after that the `split_gm` is passed to the futher steps of splitter, like replacing the `submod_1` with jitted `thunder_1`. At last the `original_split_gm` is stored in the `SubgraphInfo` and the key of `submodule_to_compiled_fns` is the original `submod_1`, so that these information can be used by other API, like benchmarking (the graph by graph benchmark needs the original GraphModule that before any changes are made by the converter)
So in the end:
split_gm in `SubgraphInfo`:
```
GraphModule(
  (wrap_body_0): GraphModule() # the module updated by converter(has thunder symbols)
  (thunder_1): GraphModule() # the module updated by spliter
)
```
original_split_gm in `SubgraphInfo`:
```
GraphModule(
  (wrap_body_0): GraphModule() # the module checkpoint calls(has torch function)
  (submod_1): GraphModule() # not jitted
)
```
If no checkpoint op exists, no deepcopy happens, the original_split_gm and split_gm are identical

Reply:
Thanks for the extra context, I think it would be helpful to have a summarized comment in code